revolv/static/js/base/accounting.js

```diff
@@ -0,0 +1,388 @@
+$(document).ready(function(){
+	//sets up initial values when the page is loaded	
+	makeAJAXcall({});
+})
+
+$(".selector").change(function(){
+	// makes an AJAX call to the accountingJSON view when the selectors are changed
+	
+	$(".added").remove();
+	var changed_selector = $(this);
+	var changed_data = {};
+
+	if (changed_selector.hasClass("start_selector")) {
+		changed_data['start_date'] = changed_selector[0].value;
+	} else if (changed_selector.hasClass("end_selector")) {
+		changed_data['end_date'] = changed_selector[0].value;
+	} else {
+		changed_data['project_choice'] = changed_selector[0].value;
+	}
+	console.log(changed_data)
+	makeAJAXcall(changed_data);
+});
+
+/**
+ * Makes the AJAX Call
+ * @param {Object} changed_data - An object with either 0 or 1 keys that maps a selector to its new value.
+ */
+function makeAJAXcall(changed_data) {
+	var csrftoken = getCookie('csrftoken');
+
+	$.ajax({
+	    url: window.JSONURL,
+	    method: "GET",
+	    data: changed_data,
+	    headers: {"X-CSRFToken": csrftoken},
+	    success:function(data, textStatus, xhr){
+	    	setValues(data);
+	    },
+	    error:function (xhr, textStatus, thrownError){
+	        alert("Sorry, something went wrong! Try again?");
+	    },
+	});
+}
+
+/**
+ * Sets the time values, creates column headers, creates row headers, and fills in table values.
+ * @param {Object} data - The JSON returned by the AJAX call.
+ */
+function setValues(data) {
+	setTimeValues(data);
+	makeColumnHeaders(data);
+	makeRowHeaders(data);
+	setTableValues(data);
+	setTotalsAndCashFlow(data);
+}
+
+/**
+ * Fills in the appropriate table cells from the given data.
+ * @param {Object} data - The JSON returned by the AJAX call.
+ */
+function setTableValues(data) {
+	placeElements(data, 'cash_in', 'donations', placeDonationElement);
+	placeElements(data, 'cash_in', 'repayments', placeROIElement);
+	placeElements(data, 'cash_in', 'adjustments', placeCashInAdjustmentElement);
+	placeElements(data, 'cash_out', 'reinvestments', placeAdminInvestmentElement);
+	placeElements(data, 'cash_out', 'adjustments', placeCashOutAdjustmentElement);
+}
+
+/**
+ * Cleanes up the String to be JQuery friendly.
+ * @param {String} text - The text to be cleaned up to be made JQuery friendly.
+ */
+function cleanUpString(text) {
+	text = text.split(' ').join('-').split(';').join('sm').split(":").join('co').split("'").join("ap").split("!").join('ex');
```

Comment:
URL encoding keeps all ascii characters except the space, but certain ascii characters(like exclamation marks and colons and semicolons and apostrophes) and the reason I'm using modified strings is so those names can go in the classnames of their respective html elements...
but now that I think about it it might just be better to use project ids